main.rb

```diff
@@ -1,39 +1,37 @@
 require_relative 'app'
 require_relative 'menu'
+require_relative 'handle_menu_options'
 
-def handle_menu_option(choice, books, people, rentals)
-  menu_actions = {
-    1 => -> { list_all_books(books) },
-    2 => -> { list_all_people(people) },
-    3 => -> { create_person(people) },
-    4 => -> { add_book(books) },
-    5 => -> { create_rental(books, people, rentals) },
-    6 => -> { list_rentals(rentals) },
-    7 => -> { exit_program }
-  }
+class LibraryApp
+  def initialize(books, people, rentals)
+    @books = books
+    @people = people
+    @rentals = rentals
+  end
+
+  def start
+    Menu.display_welcome_message
 
-  action = menu_actions[choice]
+    loop do
+      choice = Menu.display_menu_and_get_choice
 
-  action&.call
+      if (1..7).include?(choice)
+        handle_menu_option(choice, @books, @people, @rentals)
+        break if choice == 7
+      else
+        puts 'Invalid choice. Please select a valid option (1-7).'
+      end
+    end
+  end
 end
```

Comment:
- [ ] Great job refactoring your code in line with `Single-Responsibility Principle`. However, the main role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit). Anything initializing should be done inside your `app.rb` or whatever file name you used. Kindly work more on the separation of concerns.
Your `main.rb` can then look something similar to this :point_down: 

```
# *the refactor contents of main.rb file
require "./app.rb"

def main
  app = App.new()

  response = nil

  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by entering a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
    when "3"
      app.create_person()
    when "4"
      app.create_book()
    when "5"
      app.create_rental()
    when "6"
      app.list_rentals_for_person_id()
    when "7"
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

main()

```

Reply:
@Qoosim Thank you for you detailed feedback. We will work on this!